cocos/core/utils/rename-object-property.ts

```diff
@@ -0,0 +1,103 @@
+
+/**
+ * @zh
+ * 重命名对象[自身的、可枚举的](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/propertyIsEnumerable)指定属性，并且保持原本的属性顺序。
+ * @en
+ * Renames an [enumerable own](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/propertyIsEnumerable)
+ * property but also retains the original property order.
+ * 
+ * @param object @zh 原始对象。 @en The original object.
+ * @param originalPropertyKey @zh 要重命名的属性的原始键。@en Original key of the property to rename.
+ * @param newPropertyKey @zh 要重命名的属性的新键。@en New key of the property to rename.
+ * @returns
+ * @zh 若重命名成功，返回新的对象；否则返回原始对象。
+ * @en A new object which is a copy of original object if rename succeed, otherwise the original object.
```

Comment:
The function is to rename a property name, but will return a new object, it is werid.